lib/analyzer/image-extractor.ts

```diff
@@ -69,23 +69,23 @@ async function extractFromLayer(
         let buffer: Buffer | undefined;
         for (const searchAction of searchActions) {
           if (minimatch(name, searchAction.pattern, { dot: true })) {
-            if (header.type === "link" || header.type === "symlink") {
-              debug(`${header.type} '${header.name}' -> '${header.linkname}'`);
-            } else {
-              if (!buffer) {
-                buffer = await streamToBuffer(stream);
-              }
+            if (header.type === "file") {
+              // convert stream to buffer
+              buffer = buffer || (await streamToBuffer(stream));
               // initialize the files associated products dict
-              if (!result[name]) {
-                result[name] = {};
-              }
+              result[name] = result[name] || {};
```

Comment:
I think it's better to do as before (if (!result[name])) - because you're not doing reassignment, but initialization of the code